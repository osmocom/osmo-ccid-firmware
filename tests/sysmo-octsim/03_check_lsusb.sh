#!/bin/sh -e
. ./test-data

lsusb -d $USBD_TARGET -v 2>/dev/null | sed -e "s/$Bus.*Device.*:\ ID/ID/g" | sed -e "s/ID\ 1d50:6141\ OpenMoko,\ Inc.\ sysmoOCTSIM.*/ID\ 1d50:6141\ OpenMoko,\ Inc.\ sysmoOCTSIM/" | sed -e "s/^.*iProduct.*2.*sysmoOCTSIM.*$/  iProduct                2 sysmoOCTSIM/g"
