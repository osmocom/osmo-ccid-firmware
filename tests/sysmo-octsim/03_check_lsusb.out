
ID 1d50:6141 OpenMoko, Inc. 
Device Descriptor:
  bLength                18
  bDescriptorType         1
  bcdUSB               2.00
  bDeviceClass            2 Communications
  bDeviceSubClass         0 
  bDeviceProtocol         0 
  bMaxPacketSize0        64
  idVendor           0x1d50 OpenMoko, Inc.
  idProduct          0x6141 
  bcdDevice            0.00
  iManufacturer           1 sysmocom - s.f.m.c. GmbH
  iProduct                2 sysmoOCTSIM
  iSerial                 7 ecc68ade3335355320202034463715ff
  bNumConfigurations      1
  Configuration Descriptor:
    bLength                 9
    bDescriptorType         2
    wTotalLength       0x006f
    bNumInterfaces          2
    bConfigurationValue     1
    iConfiguration          3 A
    bmAttributes         0x80
      (Bus Powered)
    MaxPower              500mA
    Interface Descriptor:
      bLength                 9
      bDescriptorType         4
      bInterfaceNumber        0
      bAlternateSetting       0
      bNumEndpoints           3
      bInterfaceClass        11 Chip/SmartCard
      bInterfaceSubClass      0 
      bInterfaceProtocol      0 
      iInterface              6 CCID
      ChipCard Interface Descriptor:
        bLength                54
        bDescriptorType        33
        bcdCCID              1.10  (Warning: Only accurate for version 1.0)
        nMaxSlotIndex           7
        bVoltageSupport         7  5.0V 3.0V 1.8V 
        dwProtocols             1  T=0
        dwDefaultClock       2500
        dwMaxiumumClock     20000
        bNumClockSupported      4
        dwDataRate           6720 bps
        dwMaxDataRate      921600 bps
        bNumDataRatesSupp.      0
        dwMaxIFSD               0
        dwSyncProtocols  00000000 
        dwMechanical     00000000 
        dwFeatures       000100B0
          Auto clock change
          Auto baud rate change
          Auto PPS made by CCID
          TPDU level exchange
        dwMaxCCIDMsgLen       272
        bClassGetResponse    echo
        bClassEnvelope       echo
        wlcdLayout           none
        bPINSupport             0 
        bMaxCCIDBusySlots       8
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x02  EP 2 OUT
        bmAttributes            2
          Transfer Type            Bulk
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0040  1x 64 bytes
        bInterval               0
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x83  EP 3 IN
        bmAttributes            2
          Transfer Type            Bulk
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0040  1x 64 bytes
        bInterval               0
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x84  EP 4 IN
        bmAttributes            3
          Transfer Type            Interrupt
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0040  1x 64 bytes
        bInterval              16
    Interface Descriptor:
      bLength                 9
      bDescriptorType         4
      bInterfaceNumber        1
      bAlternateSetting       0
      bNumEndpoints           0
      bInterfaceClass       254 Application Specific Interface
      bInterfaceSubClass      1 Device Firmware Update
      bInterfaceProtocol      1 
      iInterface              8 sysmoOCTSIM DFU (Runtime)
      Device Firmware Upgrade Interface Descriptor:
        bLength                             9
        bDescriptorType                    33
        bmAttributes                        8
          Will Detach
          Manifestation Intolerant
          Upload Unsupported
          Download Unsupported
        wDetachTimeout                      0 milliseconds
        wTransferSize                     512 bytes
        bcdDFUVersion                   1.00
Device Status:     0x0000
  (Bus Powered)
